### PR TITLE
[Issue 5690][pulsar-io] Make type name of Elasticsearch sink connector configurable.

### PR DESCRIPTION
**Pull request by fantapsody**
Fixes https://github.com/apache/pulsar/issues/5690

### Motivation

The default type name cannot be compatible with Elasticsearch versions both before 6.2 and after 7.0, so this change makes it configurable.

### Modifications

Make type name configurable, and set the default value to "_doc" to make it compatible with Elasticsearch versions after 6.2 by default.

### Verifying this change

Added unit tests and local integration tests.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - **The default values of configurations: yes**
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchConfig.java [hunk 1]
```diff
@@ -52,6 +52,15 @@
     )
     private String indexName;
 
+    @FieldDoc(
+            required = false,
```
Comment (vzhikserg) [inline pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchConfig.java]: Please align the indentation level with other annotations

Comment (fantapsody) [reply pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchConfig.java]: Fixed it, thanks!

**Comment by tuteng** [comment]
LGTM
Is it convenient for you to write an example document for use, similar to this http://pulsar.apache.org/docs/en/io-cdc-debezium/#example-of-mysql

**Comment by fantapsody** [comment]
> LGTM
> Is it convenient for you to write an example document for use, similar to this http://pulsar.apache.org/docs/en/io-cdc-debezium/#example-of-mysql

Sure, I'm working on some end-to-end tests locally, and will deal with it after that.

**Comment by fantapsody** [comment]
> LGTM
> Is it convenient for you to write an example document for use, similar to this http://pulsar.apache.org/docs/en/io-cdc-debezium/#example-of-mysql

@tuteng I have added the example document, could you please review it again?

File: site2/docs/io-elasticsearch-sink.md [hunk 2]
```diff
@@ -16,21 +16,26 @@ The configuration of the ElasticSearch sink connector has the following properti
...
 | `elasticSearchUrl` | String| true |" " (empty string)| The URL of elastic search cluster to which the connector connects. |
 | `indexName` | String| true |" " (empty string)| The index name to which the connector writes messages. |
+| `typeName` | String | false | "_doc" | The type name to which the connector writes messages to. <br><br> The value should be set explicitly to a valid type name other than "_doc" for Elasticsearch version before 6.2, and left to default otherwise. |
 | `indexNumberOfShards` | int| false |1| The number of shards of the index. |
 | `indexNumberOfReplicas` | int| false |1 | The number of replicas of the index. |
 | `username` | String| false |" " (empty string)| The username used by the connector to connect to the elastic search cluster. <br><br>If `username` is set, then `password` should also be provided. |
 | `password` | String| false | " " (empty string)|The password used by the connector to connect to the elastic search cluster. <br><br>If `username` is set, then `password` should also be provided.  |
 
-### Example
+## Example
```
[the remark this answers is not in the document]
Comment (fantapsody) [reply site2/docs/io-elasticsearch-sink.md]: Thanks, the modification here is to follow the structure of https://raw.githubusercontent.com/apache/pulsar/master/site2/docs/io-cdc-debezium.md . @tuteng @Anonymitaet Could you give some suggestion here?

**Comment by tuteng** [comment]
retest this please

File: site2/docs/io-elasticsearch-sink.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (vzhikserg) [reply site2/docs/io-elasticsearch-sink.md]: OK, it is fine if you wanted to make it consistent with description of other connectors

File: site2/docs/io-elasticsearch-sink.md [hunk 3]
```diff
@@ -40,10 +45,95 @@ Before using the ElasticSearch sink connector, you need to create a configuratio
...
+    * refresh the index
+        ```bash
+            $ curl -s http://localhost:9200/my_index/_refresh
+        ``` 
+    * search documents
+        ```bash
+            $ curl -s http://localhost:9200/my_index/_search
+        ```
+        You can see the record that published earlier has been successfully written into Elasticsearch.
+        ```bash
```
Comment (vzhikserg) [inline site2/docs/io-elasticsearch-sink.md]:
```suggestion
        ```json
```

**Comment by sijie** [comment]
retest this please

**Comment by sijie** [comment]
@fantapsody can you take a look at @vzhikserg 's comment?

**Comment by fantapsody** [comment]
> @fantapsody can you take a look at @vzhikserg 's comment?

Done, thanks! @vzhikserg

**Comment by sijie** [comment]
retest this please

**Comment by sijie** [comment]
retest this please

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Pull request merged**